trillian/ctfe/config.go

```diff
@@ -78,6 +79,63 @@ func MultiLogConfigFromFile(filename string) (*configpb.LogMultiConfig, error) {
 	return &cfg, nil
 }
 
+// ValidateLogConfig checks that a signle log config is valid. In particular:
+//  - A mirror log has a valid public key and no private key.
+//  - A non-mirror log has a private, and optionally a public key (both valid).
+//  - Each of NotBeforeStart and NotBeforeLimit, if set, is a valid timestamp
+//    proto. If both are set then NotBeforeStart <= NotBeforeLimit.
+//
+// TODO(pavelkalinnikov): Return the parsed values so that we don't have to
+// parse them again after validation.
+func ValidateLogConfig(cfg *configpb.LogConfig) error {
+	if cfg.LogId == 0 {
+		return errors.New("empty log ID")
```

Comment:
Not sure if zero is invalid. It might be excluded from random generation. Did you check?

Reply:
It's a FAQ. The tree ID is [always](https://github.com/google/trillian/blob/9abd320ce4fc4ec4a39ea661b4d17ed1e46fdecc/storage/tree_id.go#L24:6) positive (luckily).

Reply:
Btw, we could check `LogId < 0` as well.

Reply:
I like strict validation but some tests that create trees might not go via the admin API.

Reply:
Yeah, one inconvenience it may cause is forcing tests to specify tree IDs in places where it's not necessary for the purposes of these test. But this PR doesn't reveal any such tests, so we are fine I think.